### DISCUSSION TITLE
yarn workspaces foreach only running on root workspace

### DISCUSSION THREAD

**Discussion by wdfinch** [yarnpkg/berry#2809] [Q&A] [answered]
Hi all I'm having an issue with the `yarn workspaces foreach` command. I simply want to run a command in every workspace. However, when I run `yarn workspaces foreach`, it seems it only runs in the root workspace where my root `package.json` exists. 

If I use the `-p` flag it seems to run in more workspaces, but it's still skipping a few. The `-A` flat seems to have no effect. 

I'm using `yarn version 2.4.1`

Here's some output to show the issue:

```
➜ yarn workspaces list                
➤ YN0000: .
➤ YN0000: apps/workspace/cadences
➤ YN0000: apps/workspace/call-flows
➤ YN0000: apps/workspace/container
➤ YN0000: packages/eslint-config
➤ YN0000: packages/event-bus
➤ YN0000: packages/module-federation
➤ YN0000: packages/prettier-config
➤ YN0000: packages/react-common
➤ YN0000: packages/react-flow-builder
➤ YN0000: packages/react-shared-api
➤ YN0000: packages/shared-proto-buff
➤ YN0000: packages/webpack-config
➤ YN0000: Done in 0s 1ms
```

```
➜  yarn workspaces foreach exec pwd     
➤ YN0000: /home/wfinch/work/dialsource1/dialsource/frontend
```

Can anyone explain what's going on here?

**Comment by merceyz** [comment, marked as answer]
I'll guess you're using Node 15 in which case you're running into https://github.com/yarnpkg/berry/pull/2255

You can fix it by running
```
yarn set version 3.0.0-rc.2
```
Note: You'll need to remove your existing plugins and import them anew

**Reply by wdfinch** [reply]
That fixed it! Thanks.
